src/include/ipc/topology.h

```diff
@@ -240,8 +240,16 @@ struct sof_ipc_comp_process {
 
 /* IPC file component used by testbench only */
 struct sof_ipc_comp_file {
+	/* These need to be the same as in above sof_ipc_comp_process */
 	struct sof_ipc_comp comp;
 	struct sof_ipc_comp_config config;
+	uint32_t size;	/**< size of bespoke data section in bytes */
+	uint32_t type;	/**< sof_ipc_process_type */
+
+	/* reserved for future use */
+	uint32_t reserved[7];
```

Comment:
I would suggest put reserved to the bottom of struct definition.

Reply:
I did it this way since currently module adapter defaults to process type IPC unless I add exception handling to module adapter.  The extra IPC data for file component is passed in the end of this struct.

I think I will try to another approach with CONFIG_LIBRARY exceptions added for SOF_COMP_FILEREAD and SOF_COMP_FILEWRITE. Not sure which is better.

Also this will be bit different with IPC4. I'm keeping this draft until I know better.